src/java.base/share/classes/sun/security/rsa/RSAPadding.java

```diff
@@ -365,20 +366,19 @@ private byte[] unpadV15(byte[] padded) throws BadPaddingException {
         byte[] data = new byte[n];
         System.arraycopy(padded, p, data, 0, n);
 
-        BadPaddingException bpe = new BadPaddingException("Decryption error");
-
         if (bp) {
-            throw bpe;
+            return Output.FAIL;
         } else {
-            return data;
+            return Output.pass(data);
```

Comment:
The Output.pass(byte[]) will create a new instance and thus make the behavior detectable.  Maybe, the Output class is not necessary, 'null' value return could be used instead.

```
- if (bp) {
-     return Output.FAIL;
- } else {
-     return Output.pass(data);
+ return bp ? null ? data;
```

Reply:
Ok, I can change it back to byte[]